DDInstall/DDInstall.psm1

```diff
@@ -0,0 +1,215 @@
+function Install-DDAgent
+{
+    param (
+        [Parameter()][ValidateSet('DotNet')][String[]]$WithAPMTracers,
+        [String] $AgentInstallURL,
+        [String] $AgentInstallerPath,
+        [Parameter(Mandatory)] [String]$ApiKey,
+        [String]$Site,
+        [String]$Tags,
+        [String]$Hostname,
+        [String]$DDAgentUsername,
+        [String]$DDAgentPassword,
+        [String]$ApplicationDataDirectory,
+        [String]$ProjectLocation,
+        [String]$InstallLogPath
+    )
+
+    $currentPrincipal = New-Object Security.Principal.WindowsPrincipal([Security.Principal.WindowsIdentity]::GetCurrent())
+    if (-not $currentPrincipal.IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator))
+    {
+        $exception = [Exception]::new("Administrator priviledges required.")
+        $PSCmdlet.ThrowTerminatingError([System.Management.Automation.ErrorRecord]::new($exception, "FatalError", [Management.Automation.ErrorCategory]::InvalidOperation, $null))
+    }
+
+    # Enable TLS 1.2
+    [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::Tls12
+
+    $installerPath = "datadog-agent.msi"
+    if ($PSBoundParameters.ContainsKey('AgentInstallerPath'))
```

Comment:
For this (and for the tracer msi below), I think you want to create a filename in the temporary directory, so that you have a fully qualified name.  
a) the suggested download below requires a fq path
b) you can (and should) clean up the downloaded MSI when done.  as written, I think this will leave 1 or 2 MSIs around in the current working directory, whereever that might be.

Reply:
This is a great catch, thank you

Reply:
great idea!